src/utils/particle/drawer.ts

```diff
@@ -121,32 +117,53 @@ export class Drawer {
                 ctx.rect(particle.position.x - radius, particle.position.y - radius, radius * 2, radius * 2);
                 break;
             case ShapeType.triangle:
-                const startX = particle.position.x - radius;
-                const startY = particle.position.y + radius / 1.66;
-                const sideLength = radius * 2;
-
-                Drawer.subDrawShape(ctx, startX, startY, sideLength, 3, 2);
+                const start: ICoordinates = {
+                    x: particle.position.x - radius,
+                    y: particle.position.y + radius / 1.66,
+                };
+
+                const side: IShapeSide = {
+                    length: radius * 2,
+                    count: {
+                        numerator: 3,
+                        denominator: 2,
+                    },
+                };
+
+                Drawer.subDrawShape(ctx, start, side);
                 break;
             case ShapeType.polygon:
                 {
-                    const startX = particle.position.x - radius / (options.particles.shape.polygon.nb_sides / 3.5);
-                    const startY = particle.position.y - radius / (2.66 / 3.5);
-                    const sideLength = radius * 2.66 / (options.particles.shape.polygon.nb_sides / 3);
-                    const sideCountNumerator = options.particles.shape.polygon.nb_sides;
-                    const sideCountDenominator = 1;
-
-                    Drawer.subDrawShape(ctx, startX, startY, sideLength, sideCountNumerator, sideCountDenominator);
+                    const start: ICoordinates = {
+                        x: particle.position.x - radius / (options.particles.shape.polygon.nb_sides / 3.5),
+                        y: particle.position.y - radius / (2.66 / 3.5),
+                    };
+                    const side: IShapeSide = {
+                        length: radius * 2.66 / (options.particles.shape.polygon.nb_sides / 3),
+                        count: {
+                            numerator: options.particles.shape.polygon.nb_sides,
+                            denominator: 1,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The key 'denominator' is not sorted alphabetically](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5012291)